accio-main/src/main/java/io/accio/main/connector/bigquery/BigQueryRecordIterator.java

```diff
@@ -116,9 +117,11 @@ private static Object getFieldValueByType(Field field, FieldValue fieldValue)
             case DATE:
                 return LocalDate.parse((String) fieldValue.getValue());
             case DATETIME:
-                return convertToMicroseconds(LocalDateTime.parse(fieldValue.getStringValue()));
+                return LocalDateTime.parse(fieldValue.getStringValue());
             case TIMESTAMP:
-                return fieldValue.getTimestampValue();
+                long microSeconds = fieldValue.getTimestampValue();
+                Instant instant = Instant.ofEpochSecond(microSeconds / 1000000, microSeconds % 1000000 * 1000);
```

Comment:
Curiously, why  `microSeconds % 1000000 * 1000`? why do we need to adjust the nano second?